### PR TITLE
Frankenstein Line Follower Code w/ CAN

### PR DESCRIPTION
**Pull request by ashwinc12**

### CODE REVIEW

File: src/main/java/frc/team4909/robot/Robot.java [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
-    m_chooser.setDefaultOption("Default Auto", kDefaultAuto);
-    m_chooser.addOption("My Auto", kCustomAuto);
-    SmartDashboard.putData("Auto choices", m_chooser);
+    Spark m_Left = new Spark(0);
+    Spark m_Right = new Spark(1);
+    // CANSparkMax m_Left = new CANSparkMax(RobotMap.leftMotorCANDevice, CANSparkMaxLowLevel.MotorType.kBrushless);
+    // CANSparkMax m_Right = new CANSparkMax(RobotMap.rightMotorCANDevice, CANSparkMaxLowLevel.MotorType.kBrushless);
+    myDrive = new DifferentialDrive(m_Left, m_Right);
+    
+    velocity = 0.5;
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// Set the initial velocity that we'll use for the two motors, for straight driving
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+    frontLeft = new DigitalInput(0);
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
We'll use three digital inputs to represent the three line-following sensors on the front.
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 3]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
-        break;
-    }
   }
+
   /**
    * This function is called periodically during operator control.
    */
   @Override
   public void teleopPeriodic() {
+    myDrive.tankDrive(velocity, velocity);
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: This line doesn't belong here. The velocity will be set later on, in one of the three if/else cases.

File: src/main/java/frc/team4909/robot/Robot.java [hunk 4]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
+    boolean frontLeftOnLine = frontLeft.get();
+    boolean frontMiddleOnLine = frontMiddle.get();
+    boolean frontRightOnLine = frontRight.get();
+    if(!frontLeftOnLine && frontRightOnLine){
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// Have we drifted to the left?
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 5]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
+      myDrive.tankDrive(velocity, velocity - 0.1);
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// Yes. Slow down the right wheel to move back towards the right.
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 6]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
+      System.out.println("Slowing Down Right");
+    }
+    else if(frontLeftOnLine && !frontRightOnLine){
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// Have we drifted to the right?
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 7]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
+      myDrive.tankDrive(velocity - 0.1, velocity);
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// Yes. Slow down the left wheel to move back towards the left.
```

File: src/main/java/frc/team4909/robot/Robot.java [hunk 8]
```diff
@@ -53,37 +68,41 @@ public void robotPeriodic() {
...
+      System.out.println("Slowing Down Left");
+    }
+    else{
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/Robot.java]: Insert comment before this line:

```
// We're either completley on the line, or completely off of the line. Drive straight.
```

**Comment by derrell** [comment]
In addition to the question of why the PR was closed, were the requested changes implemented? Silently ignored?

As a general policy, PRs should not be closed by the person who opened them, so that there aren't unanswered questions... and closing a PR (by whomever) should have a comment indicating the resolution if not resolved by merging (closed because of X).

**Comment by derrell** [comment]
Ok, that's fine. Then just comment in this PR with a link to where they ended up. That way it can be traced from the issue, to the PR, to the branch where it ended up.

**Comment by ashwinc12** [comment]
No problem.
All this code moved here: https://github.com/FRCteam4909/2019-Deep-Space/tree/VexBot

**Comment by roshanr10** [comment]
Just for future note, you can merge across branches as well (this -> VexBot), that way we can trace back where changes originate from. That way, we can resolve issues easier, if they arise.

Or better, you can merge this to testing and then pull the changes back into VexBot. Then, we have less amounts of code to test at once.

**Pull request closed without merging**